### ISSUE TITLE
Error while installing with the script

### ISSUE THREAD

**Issue by atrykush** [nextcloud/vm#1814]
## Steps to reproduce
1. Run script `sudo bash nextcloud_install_production.sh` as user ncadmin

## Expected behaviour
Script installs required services

## Actual behaviour
Script finishes with an error:

```
Installing Nextcloud...
An unhandled exception has been thrown:
OC\DatabaseException: An exception occurred while executing 'SHOW SERVER_VERSION':

Failed to connect to the database: An exception occurred in driver: SQLSTATE[08006] [7] could not connect to server: Connection refused
	Is the server running on host "localhost" (127.0.0.1) and accepting
	TCP/IP connections on port 5432? in /var/www/nextcloud/lib/private/legacy/OC_DB.php:73
Stack trace:
#0 /var/www/nextcloud/lib/private/legacy/OC_DB.php(143): OC_DB::prepare()
#1 /var/www/nextcloud/lib/private/legacy/OC_Util.php(984): OC_DB::executeAudited()
#2 /var/www/nextcloud/lib/private/legacy/OC_Util.php(964): OC_Util::checkDatabaseVersion()
#3 /var/www/nextcloud/lib/private/Console/Application.php(161): OC_Util::checkServer()
#4 /var/www/nextcloud/console.php(99): OC\Console\Application->loadCommands()
#5 /var/www/nextcloud/occ(11): require_once('/var/www/nextcl...')
#6 {main}Sorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!
sudo -u www-data php /var/www/nextcloud/occ maintenance:install --data-dir=/mnt/ncdata --database=pgsql --database-name=nextcloud_db --database-user=ncadmin --database-pass=Cwv#vrv6YLOdc6s6xW*i=cT#cw1 --admin-user=ncadmin --admin-pass=nextcloud failed
```

## Server configuration
VM on Proxmox
CPU: 2 socket 4 cores (host)
RAM: 8 Gb
HDD: 80 Gb (system), 1Tb (zfs, storage)

**Nextcloud server version:** (see your admin page)
20.0.7

**Server OS** (Ubuntu server is default)
Installed from the ubuntu-20.04.2-live-server-amd64.iso 

**How did you install the VM?** (Scripted install from master **OR** Released version)
```
$ git clone https://github.com/nextcloud/vm
$ cd vm
$ sudo bash nextcloud_install_production.sh
```

## Network
Static network configuration

**Is port 80 and/or 443 open?**
Yes

#### Log file (/var/log/nextcloud/nextcloud.log)
```
Logfile is empty
```

#### Installation log (command output)

https://0bin.net/paste/C2ISBAKM#j+2w4DYsLLSrlRNVgbvk-ShFZIIkC+X+2PNWJZH/k0j

**Comment by enoch85** [comment]
@atrykush Hey, please try again. It's probably just temporary.

I'd check your network setup, maybe that's too strict?

**Comment by atrykush** [comment]
Hi @enoch85,
Thanks for your response. 
I've tried about ten times to install with this script, and it continue to fail at this specific point.
I've seen that sometimes the Postgres DB password for generated user contains unsafe characters (i.e. '#' or '='), so I've decided to escape sequence them in BASH script, but it didn't help. Still, executing a cli command that was marked as an errorneous finishes with the same error, while I able to login with the same username/password combination with `sudo -u ncadmin psql -W -d nextcloud_db` from the console

**Comment by szaimen** [comment]
Maybe this happens because of 
postresgql gets installed and configured before this line: https://github.com/nextcloud/vm/blob/43cedf40db5a7f0a41d717ae9ae3b1617225e278/nextcloud_install_production.sh#L476
which loads the library once more into the memory: https://github.com/nextcloud/vm/blob/43cedf40db5a7f0a41d717ae9ae3b1617225e278/static/setup_secure_permissions_nextcloud.sh#L8 and thus changes the password that will be used when installing nextcloud afterwards.
But I am not sure if that is really the problem here since on all other instances that we've tested this on, this works...

**Comment by atrykush** [comment]
@szaimen @enoch85 Guys, looks like I've found something. 
I've disabled connection encryption for Postgres as well as password encryption. This allowed me to check authentication with tcpdump as passwords appear in clear. 
The password in the tcpdump looked completely different from the one specified in the CLI. I've checked and found the script takes database connection parameters from `/var/www/nextcloud/config/config.php` and not from the CLI keys, as password app. As the `/var/www/nextcloud` directory remained from the previous errorneous installation, it kept connecting with the old password. I've moved `/var/www/nextcloud` directory, removed packages, purged DB files. After that I've been able to continue installation process up till the end. 
After server restarted I was able to continue to configure it.
I think the issue could be closed now.
Thanks, guys, and have a great week!
